modules/javafx.controls/src/main/java/javafx/scene/control/Skinnable.java

```diff
@@ -51,7 +51,13 @@ public interface Skinnable {
     public ObjectProperty<Skin<?>> skinProperty();
 
     /**
-     * Sets the skin that will render this {@link Control}
+     * Sets the skin that will render this {@link Control}.
+     * <p>
+     * To ensure a one-to-one relationship between a {@code Control} and its
+     * {@code Skin}, this method may check the return value of
+     * {@link Skin#getSkinnable()} against this Skinnable,
+     * and may throw an IllegalArgumentException if it is not the same.
+     *
```

Comment:
Whether or not it is a checked exception, we should document exceptions that are thrown with `@throws`.

In this case, I see that it says the exception "might" be thrown, so if we add an `@throws` (which seems a good idea), it would need to describe the conditions under which it might be thrown.

Reply:
Yes, even if there are some places in FX that don't, the expectation is that it is listed in an @throws.
There are hundreds of examples of this in the JDK.
Also whenever you have the name of a class (etc) in the comment use {@code IllegalArgumentException)